org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeBinding.java

```diff
@@ -855,7 +882,7 @@ public boolean isFromSource() {
 
 	@Override
 	public boolean isGenericType() {
-		return this.type.isParameterized() && this.type.getTypeArguments().stream().anyMatch(TypeVar.class::isInstance);
+		return this.type.isParameterized() && getTypeArguments() != NO_TYPE_ARGUMENTS && this.type.getTypeArguments().stream().anyMatch(TypeVar.class::isInstance);
```

Comment:
Couldn't it be simplified to
```suggestion
			return this.type.isParameterized() && this.isDeclaration;
```
?